### ISSUE TITLE
Documentation for github-project should explain this is a repository

### ISSUE THREAD

**Issue by karlbrown-va** [backstage/backstage#14069]
## Feature Suggestion

Update this text: https://backstage.io/docs/features/software-catalog/well-known-annotations#githubcomproject-slug
```
The value of this annotation is the so-called slug that identifies a project on [GitHub](https://github.com/) (either the public one, or a private GitHub Enterprise installation) that is related to this entity. It is on the format <organization>/<project>, and is the same as can be seen in the URL location bar of the browser when viewing that project.
```

This text suggests that you are linking to a github project (https://docs.github.com/en/issues/planning-and-tracking-with-projects/learning-about-projects/about-projects) - however I believe the intent is to link to a repository.

## Possible Implementation

Update the markdown to explain this is actually looking for the path to a repo

## Context

I wanted to add the github-project slug for a system, but at first I thought it meant an actual 'github project' - however, this is more just an overloading, and what this slug is looking for is a repository path. (unless it also does some magic with github-projects, not sure)

**Comment by jhaals** [comment]
Yes we should change it to "identifies a repository on GitHub", would you be willing to fix this? :)

**Comment by karlbrown-va** [comment]
hi, thanks; re: deprecation, maybe it doesn't need to be deprecated, I don't know if just making github-repo as an alias would make sense as an alternative.

**Comment by rahadin996** [comment]
slug code
